### PR TITLE
Issue #11214: specified violation message for AnnotationUseStyleCheck

### PR DESCRIPTION
**Pull request by shubh220922**
This is with reference to #11214 

I have made the respective changes but is unable to wrap the violation text to the next line. It keeps showing error when ever I wrap the long violation line to the next line. 

```
[ERROR] testStyleCompactNoArray  Time elapsed: 0.008 s  <<< ERROR!
com.puppycrawl.tools.checkstyle.api.CheckstyleException: Violation message should be specified on line 20 in C:\Users\shubh\OneDrive\Desktop\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\checks\annotation\annotationusestyle\InputAnnotationUseStyleCompactNoArray.java
        at com.puppycrawl.tools.checkstyle.checks.annotation.AnnotationUseStyleCheckTest.testStyleCompactNoArray(AnnotationUseStyleCheckTest.java:210)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Violation message should be specified on line 20
        at com.puppycrawl.tools.checkstyle.checks.annotation.AnnotationUseStyleCheckTest.testStyleCompactNoArray(AnnotationUseStyleCheckTest.java:210)

[ERROR] testParensAlways  Time elapsed: 0.002 s  <<< ERROR!
com.puppycrawl.tools.checkstyle.api.CheckstyleException: Violation message should be specified on line 82 in C:\Users\shubh\OneDrive\Desktop\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\checks\annotation\annotationusestyle\InputAnnotationUseStyleWithParens.java
        at com.puppycrawl.tools.checkstyle.checks.annotation.AnnotationUseStyleCheckTest.testParensAlways(AnnotationUseStyleCheckTest.java:142)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Violation message should be specified on line 82
        at com.puppycrawl.tools.checkstyle.checks.annotation.AnnotationUseStyleCheckTest.testParensAlways(AnnotationUseStyleCheckTest.java:142)

```

Hence, I have not wrapped the line therefore, it is giving me "longer than 100 character" error.

Kindly assist as to how can I wrap the line to the next line without triggering errors.

### CODE REVIEW

**Comment by nrmancuso** [comment]
@shubh220922 `grep` project for `violation below` to see comments placed on lines above violations. Also, keep in mind that BDD comment is a pattern, so can can shorten comments a bit by replacing less important content with `.*`

**Comment by shubh220922** [comment]
@nrmancuso yes figured that out. Working on it.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleNoTrailingComma.java [hunk 1]
```diff
@@ -9,20 +9,23 @@
...
 public class InputAnnotationUseStyleNoTrailingComma
 {
-  @SuppressWarnings({"common"}) // violation
+  @SuppressWarnings({"common"}) // violation 'Annotation array values must contain trailing comma'
   public void foo() {
 
       /** Suppress warnings */
-      @SuppressWarnings({"common","foo"}) // violation
+      @SuppressWarnings(
+              {"common","foo"}) // violation 'Annotation array values must contain trailing comma'
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleNoTrailingComma.java:20]: please try to not change java code (even formatting).
you can remove extra comments, like in this case and use `// violation below .....`

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleDifferentStyles.java [hunk 2]
```diff
@@ -10,50 +10,53 @@
...
+@SomeArrays(pooches={DOGS.LEO}) // violation 'Annotation style must be 'COMPACT_NO_ARRAY''
+@SuppressWarnings({""}) // violation 'Annotation style must be 'COMPACT_NO_ARRAY''
 public class InputAnnotationUseStyleDifferentStyles
 {
 
 }
 
-@SomeArrays(pooches={DOGS.LEO}, um={}, duh={"bleh"}) // violation
+@SomeArrays(pooches={DOGS.LEO}, // violation 'Annotation style must be 'COMPACT_NO_ARRAY''
+        um={}, duh={"bleh"})
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleDifferentStyles.java:21]: use `// violation below`

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleNoParens.java [hunk 3]
```diff
@@ -81,9 +82,11 @@ class Closing3 {
 class Example9 {}
 @AnnotationWithAnnotationValue(value = @Another)
 class Example10 {}
-@AnnotationWithAnnotationValue(@Another()) // violation
+@AnnotationWithAnnotationValue(
+         @Another())// violation 'Annotation cannot have closing parenthesis'
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleNoParens.java:86]: please add new line above and place comment "// violation below ..."

**Review by romani — CHANGES_REQUESTED** [review]
please try to not change code formatting, it is weird on purpose.

it is ok to add new lines for violation comments  if that is not avoidable

items:

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleNoTrailingComma.java (hunk 1 above)
Comment (shubh220922) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleNoTrailingComma.java]: Done. Code formatting has been rewritten to original form. Violation message has been rewritten accordingly.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleDifferentStyles.java (hunk 2 above)
Comment (shubh220922) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleDifferentStyles.java]: Done. Code formatting has been rewritten to original form. Violation message has been rewritten accordingly.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleNoParens.java (hunk 3 above)
Comment (shubh220922) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleNoParens.java]: Done. Code formatting has been rewritten to original form. Violation message has been rewritten accordingly.

Comment (shubh220922) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleNoParens.java]: mvn clean verify passed without errors/failures

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleWithTrailingCommaIgnore.java [hunk 4]
```diff
@@ -30,13 +31,15 @@ public String toString() {
         };
     }
 
-    @Test3(value={"foo",}, more={"bar",}) // violation
+    @Test3(value={"foo",}, // violation 'Annotation style must be 'COMPACT_NO_ARRAY''
+            more={"bar",})
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleWithTrailingCommaIgnore.java:35]: Keep code same, put violation line above

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleNoTrailingComma.java [hunk 5]
```diff
@@ -9,20 +9,23 @@
...
       Object o = new Object() {
 
-          @SuppressWarnings(value={"common"}) // violation
+              // violation below 'Annotation array values must contain trailing comma'
+          @SuppressWarnings(value ={"common"})
           public String toString() {
 
-              @SuppressWarnings(value={"leo","herbie"}) // violation
+                  // violation below 'Annotation array values must contain trailing comma'
+              @SuppressWarnings( value={"leo","herbie"})
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleNoTrailingComma.java:28]: Do vertical alignment of comment and code

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleNoParens.java [hunk 6]
```diff
@@ -36,10 +36,11 @@ enum SON3 {
     ETHAN
 }
 
-@InputAnnotationUseStyleCustomAnnotation6() // violation
+@InputAnnotationUseStyleCustomAnnotation6// violation 'Annotation cannot have closing parenthesis'
+             ()
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleNoParens.java:40]: Keep code same, put comment above

**Review by romani — CHANGES_REQUESTED** [review]
Review your change in GitHub your self.
Apply following items to all similar cases:

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleNoParens.java (hunk 6 above)
Comment (shubh220922) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleNoParens.java:40]: Missed this line by mistake. Respective changes to be made are in progress.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleNoTrailingComma.java (hunk 5 above)
Comment (shubh220922) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleNoTrailingComma.java:30]: Vertical allignment of comments and code has been corrected.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleNoParens.java (hunk 6 above)
Comment (shubh220922) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleNoParens.java]: Done.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleExpanded.java [hunk 7]
```diff
@@ -64,24 +70,28 @@ enum E4 {
...
-    @Another({"foo", "bar"}) //compact style // violation
+    // violation below 'Annotation style must be 'EXPANDED''
+    @Another({"foo", "bar"}) //compact style
     String value1() default "";
 }
 
 @SomeArrays(pooches = {})
-@Another({}) // violation
+// violation below 'Annotation style must be 'EXPANDED''
+@Another({})
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleExpanded.java:80]: Please use comment above ONLY if there is no space to fix this comment on same line.
Sorry that I was not clear on this from the beginning.

Please apply this to all other cases like this

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleExpanded.java (hunk 7 above)
Comment (shubh220922) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleExpanded.java]: Done. Changes have been made accordingly.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleCompactNoArray.java [hunk 8]
```diff
@@ -10,27 +10,29 @@
...
 @SuppressWarnings("") //compact_no_array
 @Deprecated()
 class Dep6 {
 
 }
 
 @Deprecated
-@SomeArrays(pooches={DOGS.LEO}) // violation
-@SuppressWarnings({""}) // violation
+@SomeArrays(pooches={DOGS.LEO})// violation 'Annotation style must be 'COMPACT_NO_ARRAY''
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleCompactNoArray.java:29]: Restore spaces that are not required to be removed to comply to line length limit

**Review by romani — CHANGES_REQUESTED** [review]
Items, apply to all cases of same type:

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleCompactNoArray.java (hunk 8 above)
Comment (shubh220922) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationusestyle/InputAnnotationUseStyleCompactNoArray.java]: Done. Respective changes have been made. All spaces have been restored wherever applicable.

**Review by romani — APPROVED** [review]
Ok to merge

**Pull request merged**